### PR TITLE
Added Dropdown to choose Schedule to Add Course

### LINKED ISSUE #8: FEATURE: Form to create PS course has drop down for Course Schedules
# User Story

* As a user
* I can select Course Schedules with a drop down 
* So that I don't have to memorize id numbers to add a section to a course schedule

## Discussion

Currently the form to create a PSCourse looks like this.  You have to enter the database id of an existing course schedule.

image

What we want instead is for the field for Course Description id to be replaced with this dropdown, which is currently in the storybook as `PersonalScheduleDropdown`

image

### Acceptance Criteria

- [x] The page to create a new PSCourse (`/courses/create`) has a dropdown in place of the free form field to enter an id
- [x] Everything else about creating a PSCourse remains the same for now.

### PR DESCRIPTION
**Pull request by mageswarankk**
This commit replaces a textbox (prompt for Personal Schedule ID) with a dropdown to choose a Personal Schedule from a list with names when adding a Course to a Personal Schedule.

NOTE: There needs to be 2 schedules and the user has to click between them in the dropdown choice for it to work; it doesn't seem like (I might be wrong about this) other groups have found a workaround for this; apart from this Bug, the feature works as intended.

Deployed at https://proj-courses-mageswarankk-dev.dokku-09.cs.ucsb.edu/
at page courses/create

image

Closes #8

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
Ok, so there are so many files unrelated to the PR that have minor whitespace changes here that it's really difficult to find the actual code changes.

In some organizations, they have a zero tolerance policy for this.  I generally am chill about it when it's one or two files, but this is way over the top.

Please either:
* Fix this branch to have only the changes to the files that are needed for the new functionality, OR
* Make a new PR that has only those changes.

One way to do the first approach is this:

```
git fetch
git checkout km-pscoursecreate-dropdown
git pull origin mkm-pscoursecreate-dropdown
git reset --soft origin/main
git status
```

You should now see all of the files that were modified, showing up in "red" as files that have been modified by not yet commited.  Now, go through, and for each file that does NOT pertain to the PR, restore it to it's original version.  For example:

```
git checkout frontend/.babelrc
git status
git checkout frontend/.eslintrc.json
git status
```
etc.   Each time you do the `git checkout filename` it should restore that file to what is in the main branch, discarding the changes.  DO NOT do this for the files that actually HAVE changes that are important to the PR.   Instead, for those files, use:

```
git add frontend/src/main/components/Courses/CourseForm.js
git status
```

For each, you should see that the file turns "green" because it's going to be added to the next commit.

When done, `git status` should show you ONLY the files that you want in the PR.
Then, do a new commit:

```
git commit -m "km-add dropdown to choose schedule when adding a course"
```

Then you need to do:

```
git push origin km-pscoursecreate-dropdown -f
```

The `-f` is needed because you are "rewriting history".

Good luck!

**Comment by mageswarankk** [comment]
Yeah, I was failing an `npx eslint --max-warnings 0 .` test, and so I ran it with the `--fix` flag which went through and fixed stuff from the entire repository. All the unnecessary changes should be reverted now.

File: frontend/src/tests/pages/Courses/CoursesCreatePage.test.js [hunk 1]
```diff
@@ -91,7 +112,71 @@ describe("CoursesCreatePage tests", () => {
 
     // expect(quarterField).toHaveValue("20124");
     //expect(setQuarter).toBeCalledWith("20124"); //need this and axiosMock below?
```
Comment (pconrad) [inline frontend/src/tests/pages/Courses/CoursesCreatePage.test.js:114]: Even though this was in the original, pleas remove commented out code when you find it.
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/tests/pages/Courses/CoursesCreatePage.test.js [hunk 2]
```diff
@@ -91,7 +112,71 @@ describe("CoursesCreatePage tests", () => {
...
+    // localStorage.setItem("CourseForm-psId", 13);
+
+    expect(submitButton).toBeInTheDocument();
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => expect(axiosMock.history.post.length).toBe(1));
+
+    // expect(quarterField).toHaveValue("20124");
+    //expect(setQuarter).toBeCalledWith("20124"); //need this and axiosMock below?
```
Comment (pconrad) [inline frontend/src/tests/pages/Courses/CoursesCreatePage.test.js:179]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — CHANGES_REQUESTED** [review]
So much commented out code.

I wonder: did I not go over this enough?  

https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**